src/read_ic.c

```diff
@@ -1,1342 +1,1398 @@
+     snapshot files are read on the current CPU) we move the collisionless
+     particles such that a gap of the right size is created */
+
+    for(type = 0, nall = 0; type < 6; type++)
+    {
+        n_in_file = header.npart[type];
+        n_for_this_task = n_in_file / ntask;
+        if((ThisTask - readTask) < (n_in_file % ntask)) {n_for_this_task++;}
+
+
+        if(type == 0)
+        {
+            if(N_gas + n_for_this_task > All.MaxPartSph)
+            {
+                printf("Not enough space on task=%d for SPH particles (space for %d, need at least %lld)\n", ThisTask, All.MaxPartSph, N_gas + n_for_this_task);
+                fflush(stdout);
+                endrun(172);
+            }
+        }
+
+        nall += n_for_this_task;
+    }
+
+    if(NumPart + nall > All.MaxPart)
+    {
+        printf("Not enough space on task=%d (space for %d, need at least %lld)\n", ThisTask, All.MaxPart, NumPart + nall);
+        fflush(stdout);
+        endrun(173);
+    }
+
+    memmove(&P[N_gas + nall], &P[N_gas], (NumPart - N_gas) * sizeof(struct particle_data));
+    nstart = N_gas;
+
+
+    for(bnr = 0; bnr < 1000; bnr++)
+    {
+        blocknr = (enum iofields) bnr;
+        if(blocknr == IO_LASTENTRY) {break;}
+        if(RestartFlag == 5 && blocknr > IO_MASS) {continue;}	/* if we only do power spectra, we don't need to read other blocks beyond the mass */
+
+        if(blockpresent(blocknr))
+        {
+                /* blocks only for restartflag == 0 */
+                if(RestartFlag == 0 && blocknr > IO_U
+                   && blocknr != IO_BFLD
+#ifdef INPUT_READ_HSML
+                   && blocknr != IO_HSML
+#endif
+#ifdef EOS_CARRIES_TEMPERATURE
+                   && blocknr != IO_EOSTEMP
+#endif
+#ifdef EOS_CARRIES_ABAR
+                   && blocknr != IO_EOSABAR
+#endif
+#ifdef EOS_CARRIES_YE
+                   && blocknr != IO_EOSYE
+#endif
+#ifdef EOS_TILLOTSON
+                   && blocknr != IO_EOSCOMP
+#endif
+#if defined(HYDRO_MESHLESS_FINITE_VOLUME) && ((HYDRO_FIX_MESH_MOTION==1)||(HYDRO_FIX_MESH_MOTION==2)||(HYDRO_FIX_MESH_MOTION==3))
+                   && blocknr != IO_PARTVEL
+#endif
+#if defined(BH_GRAVCAPTURE_FIXEDSINKRADIUS)
+                   && blocknr != IO_SINKRAD
+#endif
+#if defined(CHIMES) && !defined(CHIMES_INITIALISE_IN_EQM)
+                   && blocknr != IO_CHIMES_ABUNDANCES
+#endif
+#ifdef PIC_MHD
+                   && blocknr != IO_GRAINTYPE
+#endif
+                   )
+#if defined(GDE_DISTORTIONTENSOR) && defined(GDE_READIC)
+                    if(RestartFlag == 0 && (blocknr > IO_U && blocknr != IO_SHEET_ORIENTATION))
+                        if(RestartFlag == 0 && (blocknr > IO_U && blocknr != IO_INIT_DENSITY))
+                            if(RestartFlag == 0 && (blocknr > IO_U && blocknr != IO_CAUSTIC_COUNTER))
+#endif
+                                continue;	/* ignore all other blocks in initial conditions */

```

Comment:
This is the problem -- no slug variables are read in from the initial conditions. You will have to add code similar to one of the examples above, e.g.

```
#if defined(SLUG)
                   && blocknr != IO_SLUG_STATE_INITIAL
                   && blocknr != IO_SLUG_STATE_RNG
                   && blocknr != IO_SLUG_STATE_INT
                   && blocknr != IO_SLUG_STATE_DOUBLE
#endif
```

Reply:
What this code is doing is skipping all of the variables in the initial conditions that are not listed here, so if it's not equal to any of the listed ones, then it's skipped. Since we want the slug variables to be read, we have to (confusingly) not skip them.

Reply:
`&& blocknr != IO_AGE` should also be added here.